cloudinit/ssh_util.py

```diff
@@ -213,52 +212,40 @@ def users_ssh_info(username):
     return (os.path.join(pw_ent.pw_dir, '.ssh'), pw_ent)
 
 
-def render_authorizedkeysfile(value, homedir, username):
-    if value is None:
+def render_authorizedkeysfile_paths(value, homedir, username):
+    # The 'AuthorizedKeysFile' may contain tokens
+    # of the form %T which are substituted during connection set-up.
+    # The following tokens are defined: %% is replaced by a literal
+    # '%', %h is replaced by the home directory of the user being
+    # authenticated and %u is replaced by the username of that user.
+    macros = (("%h", homedir), ("%u", username), ("%%", "%"))
+    if not value:
         value = "%h/.ssh/authorized_keys"
-    for macro, field in (("%h", homedir), ("%u", username), ("%%", "%")):
-        value = value.replace(macro, field)
-        if not value.startswith("/"):
-            value = os.path.join(homedir, value)
-    return value
+    paths = re.split(r'(?<!\\) ', value)
```

Comment:
Could we get a comment here to explain what the regex is achieving?

Reply:
@otubo can you add the above and then we can land this?

Reply:
I think we may be able to just use split() now.  @otubo was trying to account for escaping spaces in sshd_config to account for spaces in paths, but I think reality is they are just not supported.

So, the right solution might just be to use split().

Reply:
That's exactly what I was aiming for, as @smoser mentioned.
I'm gonna replace it and push again.

Reply:
Also, this is the bugzilla for that: https://bugzilla.redhat.com/show_bug.cgi?id=1642008